### PR TITLE
Rewire set methods to actual url.Values' set methods

### PR DESCRIPTION
**Pull request by alexmontecucco**
Rewire set methods to actual url.Values' set methods for:
 - setFormData
 - setQueryParams

 + a few typo fix

### CODE REVIEW

**Comment by jeevatkm** [comment]
@alexmontecucco Thank you for the PR.

File: client.go [hunk 1]
```diff
@@ -236,13 +236,13 @@ func (c *Client) SetQueryParam(param, value string) *Client {
 //
 func (c *Client) SetQueryParams(params map[string]string) *Client {
 	for p, v := range params {
-		c.QueryParam.Add(p, v)
+		c.QueryParam.Set(p, v)
```
Comment (jeevatkm) [inline client.go]: Shall we do, as same as `Request` object method?

```go
c.SetQueryParam(p, v)
```

File: request.go [hunk 2]
```diff
@@ -48,13 +48,13 @@ func (r *Request) SetHeader(header, value string) *Request {
 //
 func (r *Request) SetHeaders(headers map[string]string) *Request {
 	for h, v := range headers {
-		r.Header.Set(h, v)
+		r.SetHeader(h, v)
```
Comment (jeevatkm) [inline request.go]: This is good. Thanks.

File: request.go [hunk 3]
```diff
@@ -79,13 +79,13 @@ func (r *Request) SetQueryParam(param, value string) *Request {
 //
 func (r *Request) SetQueryParams(params map[string]string) *Request {
 	for p, v := range params {
-		r.QueryParam.Add(p, v)
+		r.SetQueryParam(p, v)
```
Comment (jeevatkm) [inline request.go]: This is good, thanks.

File: client.go (hunk 1 above)
Comment (alexmontecucco) [inline client.go]: Yes, missed, will do

**Pull request merged**